On-boarding tip

diegoreymendez opened the pull request:
Fixes [this issue](https://github.com/wordpress-mobile/WordPress-iOS-Editor/issues/466).

**Notes:**
- The popup will be shown once, then a user default will be saved to signal it's been already shown.
- The translations will need to be handled by modifying the source code when we want to change them.  Unfortunately there is no easy way around this at this time, as the strings files in glotpress are replaced by the ones generated by the localize.py.

.**Things to discuss in this PR:**
- How to make the strings be updatable in a more dynamic way.
- How to make the popup reset the user default when the app version changes.

/cc @bummytime

bummytime commented:
@diegoreymendez Just did a quick run through and the UI isn't responding after the on-boarding dialog is dismissed:

https://cloudup.com/cI0k2Y_7486

This happens on both iOS 7 & 8.

bummytime commented on WordPress/Classes/System/WPGUIConstants.h:
```diff
@@ -0,0 +1,11 @@
+
+extern const CGFloat WPAnimationDurationDefault;
+extern const CGFloat WPAnimationDurationFast;
+extern const CGFloat WPAnimationDurationFaster;
```
:+1: for putting this in a separate file. There are probably other consts we can bring here from other parts of the app.

diegoreymendez replied:
Definitely!  It'd be nice to try and standardize on these as much as possible.

meremagee commented:
Can we update the content to:

Title: Brand new editor
Text: This version of the WordPress app includes a beautiful new visual editor. Try it out by creating a new post.
Confirm/dismiss: Great, thanks!

bummytime commented on WordPress/Classes/System/WordPressAppDelegate.m:
```diff
@@ -1559,4 +1568,36 @@ - (void)didReceiveNotificationCount:(NSInteger)count
+    if (![self noBlogsAndNoWordPressDotComAccount]) {
+        
+        static NSString* const WhatsNewUserDefaultsKey = @"WhatsNewUserDefaultsKey";
+        static const CGFloat WhatsNewShowDelay = 1.0f;
```
Should this const be outside of this method? (I could go either way)

diegoreymendez replied:
Well, for static constants that are only used in a single method, and that really have no use elsewhere, I usually try to just keep it inside the method because then it means that if we remove the method, the constant will be cleaned up as well.

Please let me know if that sounds good or if you still prefer that outside of the method, and I'll take care of it.

bummytime replied:
This sounds fine.

bummytime commented on WordPress/Classes/ViewRelated/Views/WPBackgroundDimmerView.m:
```diff
@@ -0,0 +1,91 @@
+/**
+ *  @brief      The alpha value for when this view is visible.
+ */
+const CGFloat WPBackgroundDimmerViewAlphaVisible = 0.4f;
```
Should this be a public value that could be changed by the consumer?

diegoreymendez replied:
Good improvement opportunity... on it!

bummytime commented on WordPress/Classes/ViewRelated/WhatsNew/WPWhatsNewView.m:
```diff
@@ -0,0 +1,145 @@
+- (void)awakeFromNib
+{
+    NSAssert([_details isKindOfClass:[UITextView class]],
+             @"Details outlet not wired.");
```
Can I just say you are the assert king. :crown:

diegoreymendez replied:
Hahaha... yeah :cop:

bummytime commented on WordPress/Classes/ViewRelated/WhatsNew/WPWhatsNewView.m:
```diff
@@ -0,0 +1,145 @@
+    {
+        __strong __typeof(weakSelf) strongSelf = weakSelf;
+        
+        [strongSelf removeFromSuperview];
```
Looks like the ```WPBackgroundDimmerView``` is not getting removed:

![screen shot 2015-01-23 at 11 52 00 am](https://cloud.githubusercontent.com/assets/154014/5879669/bcb46c16-a2f6-11e4-8616-15d78930b386.png)

![screen shot 2015-01-23 at 11 52 23 am](https://cloud.githubusercontent.com/assets/154014/5879670/c1fd78e8-a2f6-11e4-8dd7-50e97413c7af.png)

bummytime commented:
Ok Mr. @diegoreymendez, I did a first pass. Very nice job overall. Take a peek at that ```WPBackgroundDimmerView``` issue above.

diegoreymendez commented:
@bummytime - Thanks!  I'm working on it!

@meremagee - Sure!  I'll change the texts.

diegoreymendez commented:
@bummytime - The dimmer view is now being properly removed.  However, I have found another issue: rotating is not properly supported.

I'll deal with this and report back here when it's fixed.

bummytime commented:
@diegoreymendez roger that. Just let me know when you are ready for another looksie.

diegoreymendez commented:
@bummytime - This is ready for another review.

What I did:
- Fixed the issues you found, except for one comment above to which I replied with a question.
- Fixed the lack of rotation support.
- Changed the text as requested by @meremagee.
- Run the unit tests, all good.
- Tested on both iPhone & iPad.

Important:
- Travis CI builds are failing due to a failing unit test in several of our builds, although I have no clue why that same test is not failing locally.  Also the test is not related to my changes but something about core data migrations.

bummytime commented:
@diegoreymendez I just ran through the updated code and ran it on a few devices....everything looks good to me. I did see that @thianhlu had [a few more comments](https://github.com/wordpress-mobile/WordPress-iOS-Editor/issues/466#issuecomment-71607740) even though he's supposed to be ON VACATION. :-) So, assuming he and @meremagee are cool with the copy+UI this is a :shipit:  from me.
